src/eve/codegen.py

```diff
@@ -692,16 +690,14 @@ def generic_visit(self, node: TreeNode, **kwargs: Any) -> Union[str, Collection[
                         node=node,
                     ) from e.__cause__
 
-        elif isinstance(
-            node, (collections.abc.Sequence, collections.abc.Set)
-        ) and utils.is_collection(node):
-            result = [self.visit(value, **kwargs) for value in node]
-        elif isinstance(node, collections.abc.Mapping):
-            result = {key: self.visit(value, **kwargs) for key, value in node.items()}
-        else:
-            result = self.generic_dump(node, **kwargs)
+        elif isinstance(node, (list, tuple, set, collections.abc.Set)) or (
```

Comment:
A performance thing I assume? Not sure how much you really gain as you still have to do the `collections.abc.*` instancecheck of all previous `if`/`elif` branches until one condition evaluates true.

Reply:
Well, there are not previous branches with `collections.abc.*` instance checks in this method, so it speeds up a typical case where the visited node is not a leaf node but a list or tuple collection.

Reply:
```suggestion
        elif isinstance(node, (list, tuple, collections.abc.Set)) or (
```